browser/brave_shell_integration.h

```diff
@@ -10,6 +10,8 @@
 
 namespace shell_integration {
 
+void PinDefaultShortcutForExistingUsers();
+
```

Comment:
How about having `void PinShourcut()` here and unify `win::PinToTaskbar()` and `mac::AddIconToDock()` (leaving definition in _win,mac.cc in compile-time polymorphism manner)? Then we can remove buildflag from many call sites.

```diff
- #if BUILDFLAG(IS_WIN)
-               shell_integration::win::PinToTaskbar();
- #endif
- #if BUILDFLAG(IS_MAC)
-               shell_integration::mac::AddIconToDock();
- #endif
+               shell_integration::PinShortcut();
```

Reply:
Done - d4f74cd955c95238a8f603235c62a14a5cd7ebbd Seems much better. Thanks for review!